packages/wrangler/src/config/validation-helpers.ts

```diff
@@ -673,6 +673,29 @@ const isRecord = (
 ): value is Record<string | number | symbol, unknown> =>
 	typeof value === "object" && value !== null && !Array.isArray(value);
 
+/**
+ * Ensure that all bindings in an array have unique `name` properties.
+ */
+export const validateUniqueNameProperty: ValidatorFn = (
+	diagnostics,
+	field,
+	value
+) => {
+	if (Array.isArray(value)) {
+		const names = value.map((entry) => entry.name);
+		const duplicates = names.filter((name, i) => names.indexOf(name) !== i);
```

Comment:
It might not be the case here as the number of items in this array is likely to be small (e.g. <10) but `indexOf` is often not a very efficient way to check for duplicates.
If it were larger then I would use sets or perhaps a map.